src/library_manager/lib_manager.c

```diff
@@ -310,7 +316,8 @@ struct sof_man_fw_desc *lib_manager_get_library_module_desc(int module_id)
 	uint32_t lib_id = LIB_MANAGER_GET_LIB_ID(module_id);
 	struct ext_library *_ext_lib = ext_lib_get();
 	uint8_t *buffptr = (uint8_t *)_ext_lib->desc[lib_id];
-
+	if (buffptr == NULL)
+		return -EINVAL;
 	return (struct sof_man_fw_desc *)(buffptr + SOF_MAN_ELF_TEXT_OFFSET);
```

Comment:
Something doesn't look right here. Functions expects a return of type (struct sof_man_fw_desc*) but you return -EINVAL.